pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -867,63 +945,73 @@ func (gce *GCECloud) GetAutoLabelsForPD(name string, zone string) (map[string]st
 	return labels, nil
 }
 
-// Returns a GCEDisk for the disk, if it is found in the specified zone.
-// If not found, returns (nil, nil)
-func (gce *GCECloud) findDiskByName(diskName string, zone string) (*GCEDisk, error) {
+// getDiskByName returns a GCEDisk for the disk, if it is found in the specified zone.
+// Otherwise returns a DiskNotFoundError.
+func (gce *GCECloud) getDiskByName(diskName string, zone string) (*GCEDisk, error) {
 	mc := newDiskMetricContextZonal("get", gce.region, zone)
 	disk, err := gce.manager.GetDiskFromCloudProvider(zone, diskName)
-	if err == nil {
-		return disk, mc.Observe(nil)
-	}
-	if !isHTTPErrorCode(err, http.StatusNotFound) {
-		return nil, mc.Observe(err)
+	mc.Observe(err)
+
+	if isHTTPErrorCode(err, http.StatusNotFound) {
+		return nil, &DiskNotFoundError{regional: false, diskName: diskName, diskZone: zone}
 	}
-	return nil, mc.Observe(nil)
```

Comment:
What is actually returned here?

Reply:
Before the change, when the error is NotFound we just return no disk and no error.